### DISCUSSION TITLE
Would like a automatic rule to trigger on description containing "SHELL", but not others like "SHELLY"

### DISCUSSION THREAD

**Discussion by SlimG** [firefly-iii/firefly-iii#7763] [Firefly III]
I would like a automatic rule to trigger on description containing "SHELL", but not others like "SHELLY" or "ASHELL" etc.

Essentially I would like the rule to trigger on a word surrounded by space characters or non-alphabetic characters, but it does not seem to work when I've tried these:

> " SHELL "
> ' SHELL '

Using these I get no results

**Comment by GabrielBarcia** [comment]
Do you have any other possible way to match it?

In case is useful for you, for similar problems I come up with rules like bellow to avoid conflict with other similar named descriptions.

![image](https://github.com/firefly-iii/firefly-iii/assets/4810944/d2864069-5d20-4c2b-83c0-23efbf848421)

**Reply by SlimG** [reply]
Thank you for your answer!
It is possible for me to do it that way, but it leads to a very long that needs to be updated as I go since I don't know all the variants of what follows after "SHELL "

My dream come true would be some way of invoking regex pattern matching, something like:
![image](https://github.com/firefly-iii/firefly-iii/assets/50249/97377ca2-4a63-4387-86e2-a3c0dcf33844)

Or if there are any other syntax in Firefly that allows me to simply match " SHELL " case insensitive. I don't need to match it at the beginning or end of lines as it will always appear enclosed by spaces in my Firefly III

**Reply by JC5** [reply]
Add a rule trigger "description contains 'shell'". Also add a rule trigger "description does not contain 'shelly'".
